docs/html/user_guide.rst

```diff
@@ -855,6 +855,26 @@ We are using `freeze`_ here which outputs installed packages in requirements for
 
   reqs = subprocess.check_output([sys.executable, '-m', 'pip', 'freeze'])
 
+Since pip's progress bar gets hidden when running in a subprocess, you can use
+the ``--progress-bar=json`` option for easily parsable progress information::
+
+  subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'my_package', '--progress-bar=json'])
+
+Which will give the following output after it processes each download chunk:
+
+``Progress: {"current": ######, "total": ######}``
+
+Here, ``Progress:`` indicates it is download progress. The rest of the message is JSON
+with the ``current`` number of bytes downloaded and ``total`` .whl size as key/value pairs.
+Note: ``total`` is optional and may be null.
```

Comment:
You should explain under what circumstances `total` can be null. Think about it from the reader's point of view - if their code gets a `null` value for `total`, how should they do the progress reporting? You also need to explain that the value will *either* be present on all progress lines, *or* be null on all of them. Otherwise, consumers need to ask whether they have to code defensively for the possibility of sometimes getting `null`.

(This is one reason why we avoid having a programmatic API - it's a PITA to document, and maintain that documentation, in sufficient detail 🙁)